pkg/webhook/core.oam.dev/v1alpha2/componentdefinition/mutating_handler.go

```diff
@@ -91,16 +93,18 @@ func (h *MutatingHandler) Mutate(obj *v1beta1.ComponentDefinition) error {
 		obj.Spec.Workload.Type = defRef.Name
 
 		// Create workloadDefinition which componentDefinition refers to
-		workloadDef := new(v1beta1.WorkloadDefinition)
-		err = h.Client.Get(context.TODO(), client.ObjectKey{Name: defRef.Name, Namespace: obj.Namespace}, workloadDef)
-		if err != nil {
-			if apierrors.IsNotFound(err) {
-				workloadDef.SetName(defRef.Name)
-				workloadDef.SetNamespace(obj.Namespace)
-				workloadDef.Spec.Reference = defRef
-				return h.Client.Create(context.TODO(), workloadDef)
+		if h.AutoGenWorkloadDef {
```

Comment:
we should still check before this flag, if not found , then this flag works, otherwise, we should report an error

Reply:
got it